sbin/interactive-config

```diff
@@ -0,0 +1,534 @@
+#!/bin/bash
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License version 2 as
+#  published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU General Public License for more details.
+
+BASEDIR=$(dirname $BASH_SOURCE)
+OUTPUT_DIR="saved_config"
+
+: ${FUNCTIONS_FILE="$BASEDIR/functions.sh"}
+## Load functions file
+if ! [ -e $FUNCTIONS_FILE ]
+then
+	echo "ERROR: Could not find function definitions (${FUNCTIONS_FILE})"
+	exit 1
+fi
+source $FUNCTIONS_FILE
+
+usage()
+{
+cat << EOF
+
+	interactive-config --output <output dir> --artifacts <artifacts dir> <target> <install_type>
+
+	interactive-config is a tool for user to interactively generate a usable config.
+
+	options:
+
+	--artifacts <artifacts_dir>: directory where the binary inputs for image
+	 	generation are found. The configuration script indicates what specific
+		images will be used.
+	--output <output_dir>: directory where the output configs will be used.
+		Defaults to \`pwd\`/saved_config.
+
+	<target>: The target device which will be installed.
+
+	- /dev/<x>: A physical USB device, or a network block device. This generates
+		a bootable USB installer that is capable of installing OverC onto
+		a device.
+
+	- <path>/<foo>.img: Generate a bootable image. In this mode the installer is
+		directly run to generate an image that is capable of being
+		written or flashed to a device and directly booted .. or
+		directly booted via qemu.
+
+	<install_type>: Check if this is an installer or a full image
+
+	- installer: will be a installer which will be installing on the final target
+	- full: a full image which will be running as the final system
+
+	For example:
+
+		# usb drive as sdc
+		$ interactive-config --artifacts `pwd` /dev/sdc
```

Comment:
This is closer to what we want, but even though this is an external script from the installer .. it is a single script with a lot of knowledge about the system.

I'm talking about the following:

  - in the cubeit-installer if interactive mode is turned on, it sources a set of scripts.
  - these scripts can be numbered, i.e. 0001-user-config.sh, 0002-timezone-config.sh
  - It then runs through the scripts and calls their <name>-config function (i.e. all scripts must have this routine)
  - because the scripts are sourced, they have access to the internal variables of the installer, so you don't need
    to pass things like the artifacts dir, etc.
  - the scripts can write config files to be sourced later, or modify variables that are later used by other config
    question scripts, or the installer itself .. or writing any other sort of file.
  - a new configuration can be added to the installer, by simply creating a new script in the questions subdir.

Bruce